core/src/main/java/org/jdbi/v3/core/statement/DefineNamedBindingMode.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.core.statement;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import org.jdbi.v3.core.argument.Argument;
+import org.jdbi.v3.core.argument.NullArgument;
+import org.jdbi.v3.meta.Beta;
+
+/**
+ * Selects the mode of operation of the statement {@code defineNamedBindings} feature.
+ */
+@Beta
+public enum DefineNamedBindingMode {
+    /**
+     * Bind a boolean {@true} for present values, {@false} for null values.
+     * Default behavior.
+     */
+    BOOLEAN {
+        @Override
+        Optional<Object> apply(Argument arg) {
+            return Optional.of(arg instanceof NullArgument ? false : true);
```

Comment:
doesn't this presume user-made factories will re-use our `NullArgument`?

Reply:
They should try to reuse `NullArgument` if possible.  You're right that this is a limitation, but I don't really see any other way to accomplish what I want here.

Reply:
Added a note